### ISSUE TITLE
Issue with integrated translation service

### ISSUE THREAD

**Issue by ralxxx** [johnfactotum/foliate#1394]
**Describe the bug**
Foliate allows to select bits of text and translate it (with the help of Google Translate). When a  target language different as the preselected one is chosen, there is an issue with Foliate's behavior. 
- The translation to the new language is taking place
- But the translation pop-up cannot be closed anymore
- As a consequence Foliate cannot be used anymore properly unless the application is being forced to close and reopened. 
- There is actually a workaround: If - instead of Foliate - another app is opened in the foreground for about 20 sec, the translation pop-up vanishes and Foliate is again operational.

**To Reproduce**
1. Go to any epub book
2. Click on a random text string an click the right mouse button
3. select 'translate' (now the translation for the preselected language - in my case German - is shown) but select another language - I select French)
4. Now the app is blocked.

**Expected behavior**
The translation feature is aimed to support the user at specific moments. Once the translation is done and the user clicks to text next to the translation windows, the pop-up should vanish in order to continue reading. 

**Screenshots**
![Bildschirmfoto vom 2024-09-10 15-44-43](https://github.com/user-attachments/assets/75c2ce52-0b48-43f2-8346-515b2004993d)

**Version:**
 - Foliate version:  3.1.1
 - OS/Distribution and version: Debian 12
 - Desktop environment: GNOME 43.9
 - Installation method: Flatpak

**Additional context**

**Comment by johnfactotum** [comment]
Duplicate of #1270
